From 9a39e1e4978e4e81aeb918286c7f6ed6df6660ab Mon Sep 17 00:00:00 2001
From: KreitinnSoftware <pablolabs36@gmail.com>
Date: Sat, 8 Mar 2025 22:20:54 -0300
Subject: [PATCH 10/18] bcrypt: Warn only one time BCryptCreateHash and
 BCryptDuplicateHash for avoiding some games (Like Doom 2016) constantly
 flooding logs

---
 dlls/bcrypt/bcrypt_main.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/dlls/bcrypt/bcrypt_main.c b/dlls/bcrypt/bcrypt_main.c
index 3d88430a07e..2d3a715133b 100644
--- a/dlls/bcrypt/bcrypt_main.c
+++ b/dlls/bcrypt/bcrypt_main.c
@@ -1065,6 +1065,7 @@ NTSTATUS WINAPI BCryptCreateHash( BCRYPT_ALG_HANDLE handle, BCRYPT_HASH_HANDLE *
     struct algorithm *alg = get_alg_object( handle );
     struct hash *hash;
     NTSTATUS status;
+    static int warn_once;
 
     TRACE( "%p, %p, %p, %lu, %p, %lu, %#lx\n", handle, ret_handle, object, object_len, secret, secret_len, flags );
     if (flags & ~BCRYPT_HASH_REUSABLE_FLAG)
@@ -1072,7 +1073,7 @@ NTSTATUS WINAPI BCryptCreateHash( BCRYPT_ALG_HANDLE handle, BCRYPT_HASH_HANDLE *
         FIXME( "unimplemented flags %#lx\n", flags );
         return STATUS_NOT_IMPLEMENTED;
     }
-    if (object) FIXME( "ignoring object buffer\n" );
+    if (object && !warn_once++) FIXME( "ignoring object buffer\n" );
 
     if (!alg) return STATUS_INVALID_HANDLE;
     if (!ret_handle) return STATUS_INVALID_PARAMETER;
@@ -1088,12 +1089,13 @@ NTSTATUS WINAPI BCryptDuplicateHash( BCRYPT_HASH_HANDLE handle, BCRYPT_HASH_HAND
 {
     struct hash *hash_orig = get_hash_object( handle );
     struct hash *hash_copy;
+    static int warn_once;
 
     TRACE( "%p, %p, %p, %lu, %#lx\n", handle, handle_copy, object, objectlen, flags );
 
     if (!hash_orig) return STATUS_INVALID_HANDLE;
     if (!handle_copy) return STATUS_INVALID_PARAMETER;
-    if (object) FIXME( "ignoring object buffer\n" );
+    if (object && !warn_once++) FIXME( "ignoring object buffer\n" );
 
     if (!(hash_copy = malloc( sizeof(*hash_copy) ))) return STATUS_NO_MEMORY;
 
-- 
2.55.0.windows.3

